sklearn/plot/_gridsearch_results.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+from sklearn.plot import plot_heatmap
+
+
+def _plot_1D_results(cv_results, params, ax, xlabel, ylabel, title,
+                     fmt, xtickrotation):
+    import matplotlib.pyplot as plt
```

Comment:
yea makes sense when we were using plt.gca(). 
But now that we have changed the logic to creating a new figure if ax is None, is it fair to keep it as is? This is a helper function being used by other functions so the logic remains central to all of them.
But if this doesn't sound right, we can definitely move the logic one level higher from where this function is being called.